Library/Formula/devil.rb

```diff
@@ -19,35 +51,43 @@ class Devil < Formula
     cause "invalid -std=gnu99 flag while building C++"
   end
 
-  # ./../src-IL/include/il_internal.h:230:54: error: expected ',' or '...' before 'FileName'
+  # ./../src-IL/include/il_internal.h:230:54:
+  #   error: expected ',' or '...' before 'FileName'
   # https://github.com/Homebrew/homebrew/issues/40442
   fails_with :gcc => "5"
 
-  # fix compilation issue for iluc.c
-  patch :DATA
-
   def install
     ENV.universal_binary if build.universal?
 
+    if build.head?
+      cd "DevIL/"
+      system "./autogen.sh"
+    end
+
     system "./configure", "--disable-debug",
                           "--disable-dependency-tracking",
                           "--prefix=#{prefix}",
                           "--enable-ILU",
-                          "--enable-ILUT"
+                          if build.stable? || (build.with? "ilut")
+                            "--enable-ILUT"
+                          else
+                            ""
```

Comment:
It's cleaner to make an initial `args` array, then 
```ruby
args << "--enable-ILUT" if build.stable? || build.with?("ilut")
```
Then `system "./configure", *args`.  See [`yafc.rb`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/yafc.rb) for one example.